app/models/network/conserving_producer.rb

```diff
@@ -5,18 +5,27 @@ class ConservingProducer < Technology
     extend Disableable
 
     def self.disabled?(options)
-      ! options[:flexibility]
+      options[:capping_solar_pv] == 'false'
     end
 
     def self.disabled_class
       Technology
     end
 
-    def conditional_consumption_at(frame)
-      production_at(frame)
+    def production_at(frame)
+      production = super
+      capping = (capacity * capping_percentage).abs
+
+      production > capping ? capping : production
     end
 
     def store(_frame, _amount)
     end
+
+    private
```

Comment:
The indentation is something inherited from my previous company. And I'll change it.

**Another question about `private` and style:**
The  `#######`,  surrounding `private` - which I see a lot - isn't mandatory right? Because I don't really understand the purpose of it. I can grasp the idea of "whenever a `class` is growing to large, you can more easily see where the `private` methods begin". Though, when your `class` is to large, it's doing too much.